src/include/cdb/ml_ipc.h

```diff
@@ -290,4 +290,13 @@ typedef struct MotionIPCLayer
 /* MotionIPCLayer selected */
 extern MotionIPCLayer *CurrentMotionIPCLayer;
 
+/*
+ * All the available implements of the IPC layers.
+ */
+#define IPCLAYER_IMPL_NUMBER_MAX 4
+extern int CurrentIPCLayerImplNum;
+extern MotionIPCLayer* IPCLayerImpls[IPCLAYER_IMPL_NUMBER_MAX];
```

Comment:
How about define a max enum as end in GpVars_Interconnect_Type?

Reply:
fix in "https://github.com/cloudberrydb/cloudberrydb/pull/646/commits/f6daa249c5b2b6ad647e7029644e6cd33bd4c9fd"
replace IPCLAYER_IMPL_NUMBER_MAX with INTERCONNECT_TYPE_NUM in GpVars_Interconnect_Type